openedx/adg/lms/applications/handlers.py

```diff
@@ -66,3 +71,40 @@ def update_prereq_courses_status(instance, **kwargs):  # pylint: disable=unused-
 
     if is_any_status_updated:
         hub.save()
+
+
+@receiver(post_save, sender=PersistentCourseGrade)
+def update_bu_prereq_courses_status_to_completed(instance, **kwargs):  # pylint: disable=unused-argument
+    """
+    Update bu_prereq_courses_status to COMPLETED in Application Hub
+    depending on the bu course that the learner has enrolled in
+    """
+    if instance.letter_grade and instance.passed_timestamp:
+        current_user = User.objects.get(id=instance.user_id)
+        hub = ApplicationHub.objects.filter(user=current_user).first()
+        is_any_status_updated = False
+
+        if hub.bu_prereq_courses_status == ApplicationHub.IN_PROGRESS and\
+            is_course_in_prereqs(instance.course_id, bu_prereq=True):
+            hub.bu_prereq_courses_status = ApplicationHub.COMPLETED
+            is_any_status_updated = True
+            emit_event_for_bu_courses_progress(current_user, COMPLETED)
```

Comment:
Does it check if all the bu prereq courses are completed before marking the `bu_prereq_courses_status` `COMPLETED`?